### PR TITLE
Enable auth integration tests

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR enables all the integration tests and sync the code with the `master` branch. Also, this refactors the usage of `AuthenticationError` in ballerina/http module.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/http/src/main/ballerina/src/http/auth/basic_auth_handler.bal [hunk 1]
```diff
@@ -49,14 +49,20 @@ public type BasicAuthHandler object {
     # Authenticates the incoming request with the use of the credentials passed as the Basic Auth header.
     #
     # + req - The request object.
-    # + return - Returns `true` if it is possible to authenticate with Basic Auth. Else, returns `false` or the `error` in case of an error.
-    public function process(Request req) returns boolean|error {
+    # + return - Returns `true` if it is possible to authenticate with Basic Auth. Else, returns `false` or
+    # the `AuthenticationError` in case of an error.
```
Comment (ThisaruGuruge) [inline stdlib/http/src/main/ballerina/src/http/auth/basic_auth_handler.bal]: Shall we use `http:AuthenticationError` here ?

Comment (ldclakmal) [reply stdlib/http/src/main/ballerina/src/http/auth/basic_auth_handler.bal]: Yes. I have missed that. Thanks for pointing.

File: tests/jballerina-integration-test/src/test/resources/auth/src/authservices/12_token_propagation_jwt_test.bal [hunk 2]
```diff
@@ -57,12 +57,14 @@ service passthroughService12 on listener12_1 {
         path: "/"
     }
     resource function passthrough(http:Caller caller, http:Request clientRequest) {
-        var response = nyseEP12->get("/nyseStock/stocks", message = <@untainted> clientRequest);
+        var response = nyseEP12->get("/nyseStock/stocks", <@untainted> clientRequest);
         if (response is http:Response) {
             checkpanic caller->respond(response);
         } else {
+            // TODO: Remove the below casting when new lang syntax are merged.
```
Comment (ThisaruGuruge) [inline tests/jballerina-integration-test/src/test/resources/auth/src/authservices/12_token_propagation_jwt_test.bal]: Lang syntax change will not fix this. Here we have a union type and we are trying to invoke a function on top of it, which is not allowed.

File: stdlib/http/src/main/ballerina/src/http/auth/basic_auth_handler.bal (hunk 1 above)
Comment (ldclakmal) [reply stdlib/http/src/main/ballerina/src/http/auth/basic_auth_handler.bal]: We'll address all the documentation of the error types in a different PR.

File: tests/jballerina-integration-test/src/test/resources/auth/src/authservices/12_token_propagation_jwt_test.bal (hunk 2 above)
Comment (ldclakmal) [reply tests/jballerina-integration-test/src/test/resources/auth/src/authservices/12_token_propagation_jwt_test.bal]: It is already fixed with https://github.com/ballerina-platform/ballerina-lang/pull/16812

Comment (ThisaruGuruge) [reply tests/jballerina-integration-test/src/test/resources/auth/src/authservices/12_token_propagation_jwt_test.bal]: Missed that. Thanks

**Pull request merged**